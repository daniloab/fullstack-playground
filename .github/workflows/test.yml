name: Tests

on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      - name: Setup Node.js
        uses: actions/setup-node@v2
        with:
          node-version: '14'
      - run: yarn
      - run: yarn jest