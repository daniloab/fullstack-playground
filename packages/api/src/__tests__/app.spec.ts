import { clearDbAndRestartCounters, connectMongoose, disconnectMongoose } from '../../test';
import { createGetApiCall } from '../../test/helper';

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

const url = '/api/version';

it('should get api version correctly', async () => {
  const payload = {};

  const response = await createGetApiCall({ url, payload });

  expect(response.body).toMatchSnapshot();
  expect(response.status).toBe(200);
});
