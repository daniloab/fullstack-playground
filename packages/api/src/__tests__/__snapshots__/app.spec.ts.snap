// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get api version correctly 1`] = `
Object {
  "message": "1.0.0",
  "status": "OK",
}
`;
